cmd/subnetcmd/publish.go

```diff
@@ -294,39 +294,36 @@ func getNewAlias() (string, error) {
 }
 
 func getRepoURL(reposDir string) error {
-	if repoURL == "" {
-		path := filepath.Join(reposDir, alias)
-		repo, err := git.PlainOpen(path)
+	if repoURL != "" {
+		return nil
+	}
+	path := filepath.Join(reposDir, alias)
+	repo, err := git.PlainOpen(path)
+	if err == nil {
+		// there is a repo already for this alias, let's try to figure out the remote URL from there
+		conf, err := repo.Config()
 		if err != nil {
-			app.Log.Debug(
-				"opening repo failed - alias might have not been created yet, so ignore", zap.String("alias", alias), zap.Error(err))
-		} else {
-			// there is a repo already for this alias, let's try to figure out the remote URL from there
-			conf, err := repo.Config()
-			if err != nil {
-				// TODO Would we really want to abort here?
-				return err
-			}
-			remotes := make([]string, len(conf.Remotes))
-			i := 0
-			for _, r := range conf.Remotes {
-				// NOTE: supporting only one remote for now
-				remotes[i] = r.URLs[0]
-				i++
-			}
-			repoURL, err = app.Prompt.CaptureList("Which is the remote URL for this repo?", remotes)
-			if err != nil {
-				// should never happen
-				return err
-			}
-			return nil
+			// TODO Would we really want to abort here?
+			return err
 		}
-		repoURL, err = app.Prompt.CaptureString("Provide the repository URL")
+		remotes := make([]string, len(conf.Remotes))
+		i := 0
+		for _, r := range conf.Remotes {
+			// NOTE: supporting only one remote for now
+			remotes[i] = r.URLs[0]
+			i++
+		}
+		repoURL, err = app.Prompt.CaptureList("Which is the remote URL for this repo?", remotes)
 		if err != nil {
+			// should never happen
 			return err
 		}
+		return nil
 	}
-	return nil
+	app.Log.Debug(
```

Comment:
Please add a quick comment how we got here (`err != nil` which happened pretty early)

Reply:
refactored this function -- see if it's alright with you now